interpreter/cling/lib/Interpreter/ValuePrinter.cpp

```diff
@@ -344,8 +344,452 @@ static void StreamValue(llvm::raw_ostream& o, const void* V,
+  // a final newline:
+  o << '\n';
+  functionString = o.str();
+  return functionString;
+}
+
+static std::string invokePrintValueOverload(const Value& V) {
+  Interpreter *Interp = V.getInterpreter();
+  std::stringstream printValueSS;
+  printValueSS << "cling::printValue(";
+  printValueSS << getCastValueString(V);
+  printValueSS << ");";
+  Value printValueV;
+  Interp->evaluate(printValueSS.str(), printValueV);
+  assert(printValueV.isValid() && "Must return valid value.");
+  return *(std::string*)printValueV.getPtr();
+}
+
 namespace cling {
+
+  std::string printValue(void* ptr) {
+    std::ostringstream strm;
+    if (!ptr) {
+      strm << "<<<NULL>>>";
+    } else {
+      strm << ptr;
+    }
+    return strm.str();
+  }
+
+  // Bool
+  std::string printValue(bool val) {
+    std::ostringstream strm;
+    strm << std::boolalpha;
+    strm << val;
+    strm << std::noboolalpha;
+    return strm.str();
+  }
+
+  // Chars
+  std::string printValue(char val) {
+    std::ostringstream strm;
+    if (val > 0x1F && val < 0x7F) strm << "'" << val << "'";
+    else strm << "0x" << std::hex << (int) val << std::dec;
+    return strm.str();
+  }
+
+  std::string printValue(signed char val) {
+    return printValue((char) val);
+  }
+
+  std::string printValue(const unsigned char val) {
+    return printValue((char) val);
+  }
+
+  // Ints
+  std::string printValue(short val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned short val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(int val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned int val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(long long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(unsigned long long val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(float val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(double val) {
+    std::ostringstream strm;
+    strm << val;
+    return strm.str();
+  }
+
+  std::string printValue(long double val) {
+    std::ostringstream strm;
+    strm << val << "L";
+    return strm.str();
+  }
+
+  // Char pointers
+  std::string printValue(const char *const val) {
+    std::ostringstream strm;
+    if (!val) {
+      strm << "<<<NULL>>>";
+    } else {
+      const char *cobj = val;
+      int i = 0;
+      strm << "\"";
+      while (*cobj != 0) {
+        strm << *cobj;
```

Comment:
...can we factor our the code from `printValue(unsigned char)` and reuse it here?